test/system/211-dbus.bats

```diff
@@ -87,3 +87,351 @@ teardown() {
   # shellcheck disable=SC2154
   assert [ ${#stderr_lines[@]} -eq 0 ]
 }
+
+@test "dbus: session bus inside Arch Linux" {
+  create_distro_container arch latest arch-toolbox-latest
+  expected_response="$(gdbus call \
+                         --session \
+                         --dest org.freedesktop.DBus \
+                         --object-path /org/freedesktop/DBus \
+                         --method org.freedesktop.DBus.Peer.Ping)"
+
+  run --keep-empty-lines --separate-stderr "$TOOLBOX" run --distro arch sh -c 'gdbus call \
+                                                                   --session \
+                                                                   --dest org.freedesktop.DBus \
+                                                                   --object-path /org/freedesktop/DBus \
+                                                                   --method org.freedesktop.DBus.Peer.Ping'
+
+  assert_success
+  assert_line --index 0 "$expected_response"
+
+  if check_bats_version 1.10.0; then
+    assert [ ${#lines[@]} -eq 1 ]
+  else
+    assert [ ${#lines[@]} -eq 2 ]
+  fi
+
+  # shellcheck disable=SC2154
+  assert [ ${#stderr_lines[@]} -eq 0 ]
+}
+
+@test "dbus: system bus inside Arch Linux" {
+  create_distro_container arch latest arch-toolbox-latest
+  expected_response="$(gdbus call \
+                         --system \
+                         --dest org.freedesktop.systemd1 \
+                         --object-path /org/freedesktop/systemd1 \
+                         --method org.freedesktop.DBus.Properties.Get org.freedesktop.systemd1.Manager Version)"
+
+  run --keep-empty-lines --separate-stderr "$TOOLBOX" run --distro arch sh -c 'gdbus call \
+                                                                   --system \
+                                                                   --dest org.freedesktop.systemd1 \
+                                                                   --object-path /org/freedesktop/systemd1 \
+                                                                   --method org.freedesktop.DBus.Properties.Get \
+                                                                     org.freedesktop.systemd1.Manager Version'
```

Comment:
What is the motivation behind using `org.freedesktop.DBus.Peer.Ping` to only test the session bus and `org.freedesktop.DBus.Properties.Get org.freedesktop.systemd1.Manager Version` to only test the system bus? Can't we use the same method for both types of buses?

Having the same method for both could prevent potential problems if only of those methods were to break.

Reply:
From my quick testing on Fedora 38 Workstation, `Get org.freedesktop.systemd1.Manager Version` works for both the user and system buses, but `Ping` only works for the user bus because it leads to `GDBus.Error:org.freedesktop.DBus.Error.AccessDenied` for the system bus.

We could definitely include `Get org.freedesktop.systemd1.Manager Version` for the user bus tests.

Reply:
Feel free to submit a pull request with `Get org.freedesktop.systemd1.Manager Version` for the user bus tests.